src/System.Private.ServiceModel/tests/Common/Infrastructure/src/ConditionalTestDetectors.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Infrastructure.Common
+{
+    // This class contains helper "detector" methods to detect specific
+    // conditions about the environment in which the tests are run.
+    // They are used for [ConditionalFact] and [ConditionalTheory]
+    // tests.  They are called at most once, and the return value cached.
+    internal static class ConditionalTestDetectors
+    {
+        private const string ClientCertificateSubject = "WCF Client Certificate";
+        private const string CertificateIssuer = "DO_NOT_TRUST_WcfBridgeRootCA";
+
+        public static bool IsRootCertificateInstalled()
+        {
+            try
+            {
+                ServiceUtilHelper.EnsureRootCertificateInstalled();
+                return true;
+            }
+            catch
+            {
+                return false;
```

Comment:
Reviewers: I will investigate independently how we can communicate failures here back to xunit, but I would prefer not to block this PR on that investigation.